app/models/declaration.rb

```diff
@@ -46,6 +45,36 @@ class Declaration < ApplicationRecord
     clawed_back: "clawed_back",
   }, _suffix: true
 
+  state_machine :state, initial: :submitted do
+    event :mark_eligible do
+      transition [:submitted] => :eligible
+    end
+
+    event :mark_payable do
+      transition [:eligible] => :payable
+    end
+
+    event :mark_paid do
+      transition [:payable] => :paid
+    end
+
+    event :mark_ineligible do
+      transition %i[submitted eligible payable paid] => :ineligible
```

Comment:
thinking about this, from ECF it looks like we only actually mark it as ineligible if  it's submitted?